src/docs/upgrading.dox

```diff
@@ -1,5 +1,67 @@
 /*! @page upgrading Upgrading WiredTiger applications
+@section version_310 Upgrading to Version 3.1.0
+<dl>
+
+<dt>WiredTiger on-disk log file format change</dt>
+<dd>
+The WiredTiger on-disk file format for write-ahead log files has changed
+as the log file version number was incremented. See
+<a href=https://jira.mongodb.org/browse/WT-4029>WT-4029</a> for details.
+</dd>
+
+<dt>::wiredtiger_open compatibility configuration changes</dt>
+<dd>
+The compatibility setting now takes additional options that can define
+the minimum or maximum required version of existing data files. See
+<a href=https://jira.mongodb.org/browse/WT-4098>WT-4098</a> for details.
```

Comment:
I believe "require_min" and "require_max" were introduced by different tickets, i.e. WT-4056 and WT-4098 respectively. Should we add WT-4056 here as well?